handle_multi_pay_invoice_request.go

```diff
@@ -0,0 +1,198 @@
+			// Convert invoice to lowercase string
+			bolt11 = strings.ToLower(bolt11)
+			paymentRequest, err := decodepay.Decodepay(bolt11)
+			if err != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+					"bolt11":    bolt11,
+				}).Errorf("Failed to decode bolt11 invoice: %v", err)
+
+				// TODO: Decide what to do if id is empty
+				dTag := []string{"d", invoiceInfo.Id}
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_MULTI_PAY_INVOICE_METHOD,
+					Error: &Nip47Error{
+						Code:    NIP_47_ERROR_INTERNAL,
+						Message: fmt.Sprintf("Failed to decode bolt11 invoice: %s", err.Error()),
+					},
+				}, nostr.Tags{dTag}, ss)
+				if err != nil {
+					svc.Logger.WithFields(logrus.Fields{
+						"eventId":        event.ID,
+						"eventKind":      event.Kind,
+						"paymentRequest": invoiceInfo.Invoice,
+						"invoiceId":      invoiceInfo.Id,
+					}).Errorf("Failed to process event: %v", err)
+					return
+				}
+
+				svc.PublishEvent(ctx, sub, event, resp)
+				return
+			}
+
+			invoiceDTagValue := invoiceInfo.Id
+			if invoiceDTagValue == "" {
+				invoiceDTagValue = paymentRequest.PaymentHash
+			}
+			dTag := []string{"d", invoiceDTagValue}
+
+			hasPermission, code, message := svc.hasPermission(&app, event, NIP_47_PAY_INVOICE_METHOD, paymentRequest.MSatoshi)
+
+			if !hasPermission {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":   event.ID,
+					"eventKind": event.Kind,
+					"appId":     app.ID,
+				}).Errorf("App does not have permission: %s %s", code, message)
+
+				resp, err := svc.createResponse(event, Nip47Response{
+					ResultType: NIP_47_MULTI_PAY_INVOICE_METHOD,
+					Error: &Nip47Error{
+						Code:    code,
+						Message: message,
+					},
+				}, nostr.Tags{dTag}, ss)
+				if err != nil {
+					svc.Logger.WithFields(logrus.Fields{
+						"eventId":        event.ID,
+						"eventKind":      event.Kind,
+						"paymentRequest": invoiceInfo.Invoice,
+						"invoiceId":      invoiceInfo.Id,
+					}).Errorf("Failed to process event: %v", err)
+					return
+				}
+				svc.PublishEvent(ctx, sub, event, resp)
+				return
+			}
+
+			payment := Payment{App: app, NostrEvent: nostrEvent, PaymentRequest: bolt11, Amount: uint(paymentRequest.MSatoshi / 1000)}
+			insertPaymentResult := svc.db.Create(&payment)
+			if insertPaymentResult.Error != nil {
+				svc.Logger.WithFields(logrus.Fields{
+					"eventId":        event.ID,
+					"eventKind":      event.Kind,
+					"paymentRequest": bolt11,
+					"invoiceId":      invoiceInfo.Id,
+				}).Errorf("Failed to process event: %v", insertPaymentResult.Error)
+				return
+			}
+
+			svc.Logger.WithFields(logrus.Fields{
+				"eventId":   event.ID,
+				"eventKind": event.Kind,
+				"appId":     app.ID,
+				"bolt11":    bolt11,
+			}).Info("Sending payment")
+
+			preimage, err := svc.lnClient.SendPaymentSync(ctx, event.PubKey, bolt11)
```

Comment:
This causes an issue with token refreshing since multiple requests can be made in parallel

Reply:
fixed by Adithya!